This patch consists of an upstream changes for nix 0.22 and above taken from the
upstream commit described below, plus a dependency adjustment which effectively
combines many dependency adjustsments made in seperate upstream commits.

commit f4c05081e6eed0caa82b79867bc169f9c7a1949f
Author: gwenn <gtreguier@gmail.com>
Date:   Tue Jul 13 21:32:11 2021 +0200

    Upgrade to nix 0.22

Index: rustyline/src/tty/unix.rs
===================================================================
--- rustyline.orig/src/tty/unix.rs
+++ rustyline/src/tty/unix.rs
@@ -836,7 +836,7 @@ impl Term for PosixTerminal {
         use nix::errno::Errno::ENOTTY;
         use nix::sys::termios::{ControlFlags, InputFlags, LocalFlags, SpecialCharacterIndices};
         if !self.stdin_isatty {
-            return Err(nix::Error::from_errno(ENOTTY).into());
+            return Err(ENOTTY.into());
         }
         let original_mode = termios::tcgetattr(STDIN_FILENO)?;
         let mut raw = original_mode.clone();
Index: rustyline/Cargo.toml
===================================================================
--- rustyline.orig/Cargo.toml
+++ rustyline/Cargo.toml
@@ -67,7 +67,7 @@ version = "3.1.0"
 default = ["with-dirs"]
 with-dirs = ["dirs"]
 [target."cfg(unix)".dependencies.nix]
-version = "0.18"
+version = "0"
 
 [target."cfg(unix)".dependencies.utf8parse]
 version = "0.2"
